lib/highlight.js

```diff
@@ -21,13 +21,15 @@ function highlightUtil(str, options = {}) {
   hljs.configure({ classPrefix: useHljs ? 'hljs-' : ''});
 
   const data = highlight(str, options);
+  const lang = options.lang || ''; // Maybe "data.language" should be included
```

Comment:
My understanding is that `data.language` would default to `plain` value if no lang is provided. In current behavior `wrap = true`, `plain` does get appended to class name -> `"highlight plain"`.

So, I agree with `options.lang || data.language || '';`.

Reply:
One thing I worry about is that the default behavior of Hexo with `hljs: true` will be changed.

Under the code `options.lang || data.language || ''`,
when `hljs:true` is specified with neither `lang` option nor `gutter` option,
the output of `highlight()` will be `<pre><code class="hljs plain">...`.

On the other hand, the current implementation says `<pre><code class="hljs undefined">...`.
That is, a class name `plain` is not added (adding `undefined` is a tiny bug).

Under the code `options.lang || ''` (without `data.language`),
the default output of `highlight()` will be `<pre><code class="hljs">...`.
That is, no class name is added.

So including `data.language` causes Hexo changing the default behavior with `hljs: true, line_number: false`.
Is it OK?

Reply:
> the current implementation says `<pre><code class="hljs undefined">`

which implementation are you referring to? this PR with `options.lang || data.language || ''`?

---

The main reason of having "highlight plain"  is so that "hljs: false, wrap: false" can be compatible with current theme as much as possible. I just worry that some theme might use `"plain"` class.

> So including data.language causes Hexo changing the default behavior with `hljs: true, line_number: false`.

I don't foresee any breaking change for adding a new class (i.e. "plain"), even when it's not used in the newer syntax.

---

In summary, without `data.language`, "plain" class is removed when "hljs: false, wrap: false" which is a possible breaking change. Having an unused "plain" class in "hljs: true" is not a breaking change.

Reply:
@curbengh Thank you for explaining.  I didn't understand enough.
Now I understand that `data.language` is necessary to keep compatibility with current themes and adding a class `plain` will not break current themes.

I'll update the code soon.